Assets/Scripts/GameScene/BoardGenerator.cs

```diff
@@ -1,38 +1,50 @@
-using System.Collections;
-using System.Collections.Generic;
+using System;
 using UnityEngine;
 
 public class BoardGenerator : MonoBehaviour
 {
-    [SerializeField] private float width;
-    [SerializeField] private float height;
-    [SerializeField] private FloorTile tile;
-    
-    
-    // Start is called before the first frame update
-    void Start()
+    public static event Action<BoardGenerator> BoardDidLoad;
+    [SerializeField] private Board board; 
+    public Vector3[,] boardData { get; private set; }
+
+    private void Awake()
     {
-        GenerateBoard();
+        GenerateBoardData();
+        InstantiateBoard(boardData);
+        BoardDidLoad?.Invoke(this);
     }
-
-    private void GenerateBoard()
+    
+    private void GenerateBoardData()
     {
-        float initialX = (-width / 2) + (tile.width / 2);
-        float initialZ = (-height / 2) + (tile.height / 2);
-
+        int width = board.width;
+        int height = board.height;
+        FloorTile tile = board.tile;
+        boardData = new Vector3[width, height];
```

Comment:
bad naming, this is NOT the board's data, this is simply the board's tile positions, rename this to `tilePositions` or `boardPositions`